### ISSUE TITLE
Layout/SpaceInsideArrayLiteralBrackets incorrectly adds a rubocop:todo with --disable-uncorrectable 

### ISSUE THREAD

**Issue by bbugh** [rubocop/rubocop#10388]
When running with `--auto-correct --disable-uncorrectable`, a `rubocop:todo` is unnecessarily added. When running rubocop again, it complains with `Unnecessary disabling of Layout/SpaceInsideArrayLiteralBrackets`

--------

## Expected behavior

Since `Layout/SpaceInsideArrayLiteralBrackets` can be auto fixed, it should not output a todo. 

## Actual behavior

1\. Given a file `test.rb`:

```ruby
# frozen_string_literal: true

puts [ :something ]
```

2\. When running with the default rubocop config (and no other configs in the parent folder tree), it incorrectly adds a TODO for the second bracket on the line:

```
» rubocop --auto-correct --disable-uncorrectable
Inspecting 1 file
C

Offenses:

example.rb:3:9: C: [Corrected] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
puts [ :something ]
        ^
example.rb:3:20: C: [Todo] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
puts [ :something ]
                   ^

1 file inspected, 2 offenses detected, 2 offenses corrected
```

The file then looks like this:

```ruby
# frozen_string_literal: true

puts [:something] # rubocop:todo Layout/SpaceInsideArrayLiteralBrackets
```

3\. Then running same command again results in more changes because of the unnecessary todo:

```
» rubocop --auto-correct --disable-uncorrectable
Inspecting 1 file
W

Offenses:

example.rb:3:19: W: [Corrected] Lint/RedundantCopDisableDirective: Unnecessary disabling of Layout/SpaceInsideArrayLiteralBrackets.
puts [:something] # rubocop:todo Layout/SpaceInsideArrayLiteralBrackets
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```

Now the file is correct:

```ruby
# frozen_string_literal: true

puts [:something]
```

## Steps to reproduce the problem

Same as "actual" but I'll outline the simple steps:

1. Given a file `test.rb`:
    ```ruby
    # frozen_string_literal: true
    
    puts [ :something ]
    ```
2. `rubocop --auto-correct --disable-uncorrectable`
3. `rubocop`
4. Observe that `Lint/RedundantCopDisableDirective` is unexpectedly reported

## RuboCop version

```
rubocop -V
1.25.0 (using Parser 3.1.0.0, rubocop-ast 1.15.1, running on ruby 3.1.0 arm64-darwin21)
  - rubocop-rails 2.13.2
  - rubocop-rspec 2.8.0
```

**Comment by issyl0** [comment]
It looks like this is marking the rightmost space between the square bracket as `todo` because it does not detect that it is autocorrectable.

```
❯ bundle exec rubocop test.rb
Offenses:

test.rb:3:7: C: [Correctable] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
puts [ :something ]
         ^
test.rb:3:18: C: Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
puts [ :something ]
                             ^
1 file inspected, 2 offenses detected, 1 offense auto-correctable
```

However, the first run of `--auto-correct` fixes the two bracket spacing offenses in one go, but it only reports that it fixed one (the one previously marked [Correctable]):

```
❯ bundle exec rubocop test.rb --auto-correct
test.rb:3:7: C: [Corrected] Layout/SpaceInsideArrayLiteralBrackets: Do not use space inside array brackets.
puts [ :something ]
      ^
1 file inspected, 1 offense detected, 1 offense corrected
```

Not sure if this is a problem with `Layout/SpaceInsideArrayLiteralBrackets` cop itself, or the `--disable-uncorrectable` option.
